### DISCUSSION TITLE
How to add a image/watermark to the entire page

### DISCUSSION THREAD

**Discussion by ekjuanrejon** [QuestPDF/QuestPDF#139] [General]
Hello, I have a custom image that should be placed unto the entire page. How can i set the image as the background of the page?

Thanks

**Comment by MarcinZiabek** [comment]
Hello 😁 You can use the `Layers` element to put stuff behind or after the main content. However, I have noticed that creating a real watermark that covers entire page is slightly more difficult within current API (due to definition of header/content/footer slots). I have already improved the API that will be available very soon within the 2022.03 version.

Please take a look. I hope that we can discuss the API clarity and naming:

```csharp
document.Page(page =>
{
    page.Size(PageSizes.A4);
    page.Margin(1, Unit.Inch);
    page.DefaultTextStyle(TextStyle.Default.FontSize(16));

    page.Foreground() // <- new slots, Foreground and Background
        .AlignMiddle()
        .AlignCenter()
        .Text("Watermark")
        .FontSize(64)
        .FontColor(Colors.Blue.Lighten3);
    
    page.Header().Text("Background and foreground").Bold().FontColor(Colors.Blue.Medium).FontSize(24);
    
    page.Content().PaddingVertical(25).Column(column =>
    {
        column.Spacing(25);

        foreach (var i in Enumerable.Range(0, 100))
            column.Item().Background(Colors.Grey.Lighten2).Height(75);
    });
});
```

I am still not sure how to name the methods. Right now, I have Background and Foreground. In Word, it is called Watermark as far as I know. I want to support both layers behind and in front of the main content. Do you have any ideas?

**Reply by ekjuanrejon** [reply]
I have a png image which I want to use as a background. is it possible?

**Reply by ekjuanrejon** [reply]
might have to use two libraries
questpdf 
itextsharp

**Reply by MarcinZiabek** [reply]
You can put anything as a watermark, including images 😁 Just use the same API as anywhere else. 
In fact, your watermak may even contain tables and other complex structures, though I don't see much value in it...

**Reply by MarcinZiabek** [reply]
The functionality is also already available with the QuestPDF 2022.3 release.

**Reply by ekjuanrejon** [reply]
i see 2022.3 just got released in nuget

**Comment by ekjuanrejon** [comment]
Foreground puts everything on top of header, content and footer. how can i move header and content and footer on top?

**Reply by MarcinZiabek** [reply]
You can use also the Layers element inside the Content slot.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
